### ISSUE TITLE
[New Service]: Amazon Q

### ISSUE THREAD

**Issue by silvaalbert** [hashicorp/terraform-provider-aws#36464]
### Description

Add support for Amazon Q

### Requested Resource(s) and/or Data Source(s)

* aws_q_ ...

### References

https://docs.aws.amazon.com/amazonq/latest/api-reference/API_Operations.html

### Would you like to implement a fix?

Yes

**Comment by madhavvishnubhatta** [comment]
Creating a new resource - aws_qbusiness_index - I will work on this.
API Calls:
https://docs.aws.amazon.com/amazonq/latest/api-reference/API_CreateIndex.html
https://docs.aws.amazon.com/amazonq/latest/api-reference/API_DeleteIndex.html
https://docs.aws.amazon.com/amazonq/latest/api-reference/API_UpdateIndex.html

**Comment by sportlane** [comment]
I have implmented all the Q business resources:
https://github.com/sportlane/terraform-provider-aws/tree/f-aws_qbusiness-resources

I am waiting for https://github.com/hashicorp/terraform-provider-aws/pull/35249 PR to be merged so I can submit all the rest.

**Comment by sportlane** [comment]
```aws_qbusiness_app``` has been converted to use the new framework in PR #35249 . Once it is approved I will work on converting rest of the resources.

**Comment by kpx-dev** [comment]
awesome work @sportlane , any help needed with the PR?

**Comment by sportlane** [comment]
PR #35249 is ready for review and merge. I do not know what else I can do.

**Comment by stefenAMZ** [comment]
Any new movement on this? With Q growing in popularity, would be nice to have this in our provider
